internal/impl/io/output_http_server.go

```diff
@@ -386,39 +412,107 @@ func (h *httpServerOutput) streamHandler(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	if h.conf.StreamFormat == StreamFormatEventSource {
+		// Set headers for SSE
+		w.Header().Set("Content-Type", "text/event-stream")
+		w.Header().Set("Cache-Control", "no-cache")
+		w.Header().Set("Connection", "keep-alive")
+
+		if h.conf.CORS.Enabled {
+			if len(h.conf.CORS.AllowedOrigins) > 0 {
+				// Use configured origins
+				origin := r.Header.Get("Origin")
+				for _, allowed := range h.conf.CORS.AllowedOrigins {
+					if allowed == "*" {
+						w.Header().Set("Access-Control-Allow-Origin", "*")
+						break
+					}
+					if origin == allowed {
+						w.Header().Set("Access-Control-Allow-Origin", origin)
+						break
+					}
+				}
+			}
+		}
+	}
+
 	ctx, done := h.shutSig.SoftStopCtx(r.Context())
 	defer done()
 
+	var heartbeatTicker *time.Ticker
+	if h.conf.StreamFormat == StreamFormatEventSource && h.conf.Heartbeat > 0 {
+		heartbeatTicker = time.NewTicker(h.conf.Heartbeat)
+		defer heartbeatTicker.Stop()
+	}
```

Comment:
Consider declaring the heartbeat channel once:

```suggestion
	var heartbeatChan <-chan time.Time
	if h.conf.StreamFormat == StreamFormatEventSource && h.conf.Heartbeat > 0 {
		heartbeatTicker = time.NewTicker(h.conf.Heartbeat)
		defer heartbeatTicker.Stop()
		heartbeatChan = heartbeatTicker.C
	}
```